1635 Clear trash projects quick fix

Jonathanko52 opened the pull request:
Fixes #1635 

### What changes did you make?

When users view their saved projects, the fetch function now includes an imbedded delete function that scans through the fetched projects and deletes any that are in the trash and 30 days old.

### Why did you make the changes (we will use this info to test)?

The idea is for TDM to automatically delete all projects in the database that have been in the trash for 30 days. For now, we don't have that built into the back end, so we have the users automatically check and clean their own trash bin whenever they look at their projects. This does present a problem is the user trashes a bunch of projects and never logs in again, leaving their trash uncleaned, but this should work for now.

entrotech commented on client/src/hooks/useGetProjects.js:
```diff
@@ -5,12 +5,23 @@ const useProjects = handleError => {
+          let numberOfDaysSinceCreation =
+            Math.abs(new Date() - new Date(cur.dateCreated)) /
+            (1000 * 60 * 60 * 24);
+          if (numberOfDaysSinceCreation >= 30) {
```
Issue says deletion should happen after 90 days, not 30.

entrotech commented on client/src/hooks/useGetProjects.js:
```diff
@@ -5,12 +5,23 @@ const useProjects = handleError => {
+            Math.abs(new Date() - new Date(cur.dateCreated)) /
+            (1000 * 60 * 60 * 24);
+          if (numberOfDaysSinceCreation >= 30) {
+            projectService.del(cur.id);
```
This should be called async, i.e. `async projectService.del(cur.id);`

entrotech reviewed (changes requested):
The issue also calls for changing the message in the delete dialog to tell the user that the project will be permanently deleted after 90 days.

Jonathanko52 commented:
Added text to delete modal, changed the date check to 90 days (also changed it to track "days since trashed" instead of "days since creation". Oops.) And extended the delete into its own separate, asynchornous function.

![Screenshot 2024-05-01 at 7 25 07 PM](https://github.com/hackforla/tdm-calculator/assets/30099154/fd410a89-6e3b-464a-9b78-8cf0000dcbb1)
